src/commands/AbstractNewCamelProjectCommand.ts

```diff
@@ -0,0 +1,176 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License", destination); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+import * as path from "path";
+import { TaskScope, Uri, commands, window, workspace } from "vscode";
+import { CamelExportJBangTask } from "../tasks/CamelExportJBangTask";
+import { arePathsEqual, getCurrentWorkingDirectory } from "../requirements/utils";
+
+export abstract class AbstractNewCamelProjectCommand {
+
+	async create() {
+		const runtime = await this.getRuntime();
+		const input = await this.askForGAV();
+		if (runtime && input) {
+
+			// Uses the user selected folder otherwise default to the root workspace folder
+			const outputFolder = await this.showDialogToPickFolder();
+			if (!outputFolder) {
+				await window.showErrorMessage('Camel project creation canceled or invalid folder selection');
+				return;
+			}
+
+			// If the chosen ouputh folder is diferent from the first folder from the current workspace warns the user about
+			// potentially deleting files in the selected folder.
+			// Executing the command from the same folder does not delete files.
+			const cwd = getCurrentWorkingDirectory();
+			if (cwd && !arePathsEqual(cwd, outputFolder.fsPath)){
+				const userChoice = await this.confirmDestructiveActionInSelectedFolder(outputFolder.fsPath);
+
+				if (userChoice === undefined){
+					await window.showInformationMessage('Camel project creation canceled');
+					return;
+				}
+			}
+
+			await new CamelExportJBangTask(TaskScope.Workspace, input, runtime, outputFolder.fsPath).execute();
+
+			// if not exist, init .vscode with tasks.json and launch.json config files
+			await workspace.fs.createDirectory(Uri.file(path.join(outputFolder.fsPath, '.vscode')));
+			for (const filename of ['tasks', 'launch']) {
+				await this.copyFile(`../../../resources/${runtime}/${filename}.json`, path.join(outputFolder.fsPath, `.vscode/${filename}.json`));
+			}
+
+			// open the newly created project in a new vscode instance
+			await commands.executeCommand('vscode.openFolder', outputFolder, true);
+
+		}
+	}
+
+	abstract getRuntime(): Promise<string|undefined>;
```

Comment:
I think we need to decalre it as async to have a better typing and ensure that await is called.
For instance, in camel.project.command.test.ts it is not. It is causing the title of the test to be incorrect `Should validate Create a Camel Project command GAV validation of [object Promise]`, see https://github.com/camel-tooling/camel-lsp-client-vscode/actions/runs/12173733325/job/33963279530?pr=1994#step:13:123